Fix WebAssembly Csound build

kunstmusik opened the pull request:
Csound wasm-bin and browser builds were not up to date with latest changes to Csound API. 

**_NOTE: Currently set to draft PR for shared development and testing and is not yet ready for review._** 

This PR should also fix issues with github actions build/testing of Csound wasm build. A bug was found that the locally built csound/wasm-bin was not being used with the build/test of csound/browser, so an older version of wasm-bin was used for testing.

kunstmusik committed:
fix: updating wasm build to work with updated API changes

kunstmusik committed:
fix: incorrect mode for compileCSD used for test

joachimheintz commented:
great that you could sort out the main issues!
i am wondering whether this version could also be set as "Csound 7 
Alpha" in ide.csound.com for testing?

On 31/08/2025 18:51, Steven Yi wrote:
> Csound wasm-bin and browser builds were not up to date with latest 
> changes to Csound API.
> 
> */NOTE: Currently set to draft PR for shared development and testing and 
> is not yet ready for review./*
> 
> This PR should also fix issues with github actions build/testing of 
> Csound wasm build. A bug was found that the locally built csound/wasm- 
> bin was not being used with the build/test of csound/browser, so an 
> older version of wasm-bin was used for testing.
> 
> ------------------------------------------------------------------------
> 
> 
>         You can view, comment on, or merge this pull request online at:
> 
> https://github.com/csound/csound/pull/2249 <https://github.com/csound/ 
> csound/pull/2249>
> 
> 
>         Commit Summary
> 
>   * c2bc421 <https://github.com/csound/csound/pull/2249/commits/
>     c2bc421f3121599c9235a09ad3e141042d5362f4> fix: updating wasm build
>     to work with updated API changes
> 
> 
>         File Changes
> 
> (16 files <https://github.com/csound/csound/pull/2249/files>)
> 
>   * *M* Top/main.c <https://github.com/csound/csound/pull/2249/
>     files#diff-07f83befa97d8de3134a819823bc78ed1ed0162344d2cd1f66a6483a6c7f21bf> (4)
>   * *M* wasm/browser/README.md <https://github.com/csound/csound/
>     pull/2249/
>     files#diff-9711bec596f2d2b2f3d292e48d78f8e7003d09b9915313292fa998dacc697401> (317)
>   * *M* wasm/browser/externs.js <https://github.com/csound/csound/
>     pull/2249/
>     files#diff-77d5346ce1c5e1147c57422f2df7dc7fd526564e803b65f81ef6a24e2ceac376> (3)
>   * *M* wasm/browser/index.d.ts <https://github.com/csound/csound/
>     pull/2249/files#diff-
>     a69cd36e905b4345152a2c6abc0eefdb8ba73300a60d85bca8fd05e768a162cc> (6)
>   * *M* wasm/browser/src/libcsound.js <https://github.com/csound/csound/
>     pull/2249/
>     files#diff-1b697a0444cbad3a4e952ec550953e197e810bd3ad14b9073f10e0d910a12f6a> (6)
>   * *M* wasm/browser/src/modules/performance.js <https://github.com/
>     csound/csound/pull/2249/files#diff-
>     d006a54ea73dba18ea335797676f439da6265774016230e4859a760bfad6b102> (25)
>   * *M* wasm/browser/tests/channel_test.html <https://github.com/csound/
>     csound/pull/2249/
>     files#diff-8ea6a482e52b3751233508d63b84c7df4acff8e67398512a371cb878d6ac4184> (2)
>   * *M* wasm/browser/tests/diskin.html <https://github.com/csound/
>     csound/pull/2249/files#diff-
>     e8bdee3e23241e69d88d0cd8f7cafd7d2695d960d4b665ae4f4f7bede4894435> (2)
>   * *M* wasm/browser/tests/hrtf-test.html <https://github.com/csound/
>     csound/pull/2249/
>     files#diff-6a964197299c8d2ed75cb9536d486ecb51da3925ec1ecd746ca4fb81aa6e6a7d> (2)
>   * *M* wasm/browser/tests/listener_test.html <https://github.com/
>     csound/csound/pull/2249/
>     files#diff-272c9c49a5bf5ab6ce9fdaf685357d6d128e05f585b3ab09294a12fc1eb11598> (2)
>   * *M* wasm/browser/tests/mic_test.html <https://github.com/csound/
>     csound/pull/2249/
>     files#diff-710cb52aa356812c521332b2183a23af3957084eb61a3b3df007235c63fda098> (2)
>   * *M* wasm/browser/tests/midi-test.html <https://github.com/csound/
>     csound/pull/2249/
>     files#diff-8fca7254e646726c1d5c03bb57a372320ce5f86d05c21f785fc949071e80d778> (2)
>   * *M* wasm/browser/tests/tests.js <https://github.com/csound/csound/
>     pull/2249/
>     files#diff-495ed3c7fe6ca284adff9b58d0b4ccc37c457ba7d69bab2c4ee41f14ee85572c> (24)
>   * *M* wasm/browser/tests/tools-no-audio.html <https://github.com/
>     csound/csound/pull/2249/
>     files#diff-9115a0c4b76f18baa6ac6ebee13891284806997111020687d9d8370d909b8752> (2)
>   * *M* wasm/src/csound.nix <https://github.com/csound/csound/pull/2249/
>     files#diff-2063c9f7e3fd75272005fdab4649f27b38a0aa3512a8831a6a7c2312ebdf3c28> (8)
>   * *M* wasm/src/csound_wasm.c <https://github.com/csound/csound/
>     pull/2249/
>     files#diff-48a143ac6411e8e21e27680ac9d68c2bf78ea90e672df11468dc3e0944510b0a> (4)
> 
> 
>         Patch Links:
> 
>   * https://github.com/csound/csound/pull/2249.patch <https://
>     github.com/csound/csound/pull/2249.patch>
>   * https://github.com/csound/csound/pull/2249.diff <https://github.com/
>     csound/csound/pull/2249.diff>
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ 
> csound/csound/pull/2249>, or unsubscribe <https://github.com/ 
> notifications/unsubscribe-auth/ 
> AAQYHKVUZB777TMXJUSJ4KL3QMR2TAVCNFSM6AAAAACFINKDYOVHI2DSMVQWIX3LMV43ASLTON2WKOZTGM3TANBYG4YDCNI>.
> You are receiving this because you are subscribed to this thread.Message 
> ID: ***@***.***>
>

vlazzarini commented:
Csound beta?

hlolli reviewed (dismissed):
lgtm, we pair on the rest :)

vlazzarini commented:
Still work in progress?  Also needs merge from develop so CI can pass

kunstmusik commented:
Yes, this is still marked as a draft so it is not yet ready for review. I have the build passing tests but I'm working with Hlodver today to iron out some things.

kunstmusik committed:
Merge remote-tracking branch 'origin/develop' into fix-wasm-build

kunstmusik committed:
fix: cachix token and updated to v5 of checkout

kunstmusik committed:
fix: re-enable test reporting

kunstmusik committed:
fix: link to local wasm-bin

kunstmusik committed:
fix: missing names for implementations

kunstmusik committed:
chore: bump package version

kunstmusik committed:
chore: bump package version

hlolli reviewed (approved):
lgtm

kunstmusik commented:
Work completed:

* Updated wasm-bin build to latest Csound API
* Updated browser build and API for the above
* Fixed test build for missing yarn link and re-enabled test result uploading and verification (tests were being run but not evaluated for pass/fail and thus always showing success; now fixed)
* Released @csound/wasm-bin and @csound/browser on NPM with versions 7.0.0-beta1

vlazzarini commented:
Well done for the fast turnaround. 
I am wondering whether we ought to include a csound-wasm package in the beta releases but have no idea how it should look like content-wise.

joachimheintz commented:
fantastic!
any chance to put it as option on ide.csound.com?

On 01/09/2025 18:20, Steven Yi wrote:
> *kunstmusik* left a comment (csound/csound#2249) <https://github.com/ 
> csound/csound/pull/2249#issuecomment-3242862084>
> 
> Work completed:
> 
>   * Updated wasm-bin build to latest Csound API
>   * Updated browser build and API for the above
>   * Fixed test build for missing yarn link and re-enabled test result
>     uploading and verification (tests were being run but not evaluated
>     for pass/fail and thus always showing success; now fixed)
>   * Released @csound/wasm-bin and @csound/browser on NPM with versions
>     7.0.0-beta1
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ 
> csound/csound/pull/2249#issuecomment-3242862084>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ 
> AAQYHKQM7ACZ5UNHKUH3VGT3QRW45AVCNFSM6AAAAACFINKDYOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTENBSHA3DEMBYGQ>.
> You are receiving this because you commented.Message ID: <csound/csound/ 
> ***@***.***>
>

kunstmusik commented:
@joachimheintz: @hlolli was going to take a look for web-ide. 
@vlazzarini: I don't think we'll need a package as part of our release as most developers will use NPM with their build systems to use the package, but after building we can do `yarn pack` to create a local zip file of the package.
